### ISSUE TITLE
Save All Attachments not working

### ISSUE THREAD

**Issue by jurf** [flathub/org.gnome.Geary#5]
Geary is missing filesystem permissions for this (though maybe portals would be a better idea?).

**Comment by mjog** [comment]
We're using GtkNativeFileChooser, which should be using the portal. What are you seeing?

**Comment by jurf** [comment]
Corrected the title, it only does it when trying to save files, this is the screen I get:

![snimka obrazovky z 2018-02-01 17-42-58](https://user-images.githubusercontent.com/7472425/35690915-720c6e3c-0777-11e8-9a63-14b9f75a577d.png)

**Comment by TingPing** [comment]
Anything printed in the terminal? Gtk usually complaints if it fails to use the portal and falls back to a normal dialog.

**Comment by jurf** [comment]
Got it!
```
(geary:2): Gtk-WARNING **: GTK_FILE_CHOOSER_ACTION_SELECT_FOLDER is not supported by GtkFileChooserNativePortal
```
It only happens if you press ’Select all’ before pressing ‘Save’. Also `xdg-desktop-portal` crashes if I do anything file manager related after that, and there are tonnes of AVC declines (F27). Weird.

**Comment by mjog** [comment]
Thanks for tracking that down. It looks like a limitation with the current portal implementation and/or GTK+'s GtkFileChooserNative implementation: https://bugzilla.gnome.org/show_bug.cgi?id=785884

Geary [bug 793410](https://bugzilla.gnome.org/show_bug.cgi?id=793410) filed to cover finding a workaround, [xdg-desktop-portal isue #158](https://github.com/flatpak/xdg-desktop-portal/issues/158) to track the required portal additions.

**Comment by mjog** [comment]
Fixed with xdg-desktop-portal 1.7.1
